### PR TITLE
Jv stream guthub practicemy hw

### PR DESCRIPTION
**Pull request by VolodymyrKachan**

### CODE REVIEW

File: src/main/java/practice/StreamPractice.java [hunk 1]
```diff
@@ -14,7 +19,15 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(n -> Arrays.stream(n.split(","))
```
Comment (valentyn-chepiha) [inline src/main/java/practice/StreamPractice.java:23]: use magic value for ","

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -75,6 +104,10 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream()
+                .filter(new CandidateValidator())
+                .sorted(new CandidateValidator.CandidateComparator())
```
Comment (valentyn-chepiha) [inline src/main/java/practice/StreamPractice.java:109]:
```suggestion
                .sorted()
```
try without comparator

**Review by valentyn-chepiha — CHANGES_REQUESTED** [review]

File: src/main/java/practice/StreamPractice.java (hunk 1 above)
Comment (fifahater) [reply src/main/java/practice/StreamPractice.java:23]: I think that putting "," into constants is not a good idea, and so from the split method it becomes clear what it is

**Review by valentyn-chepiha — APPROVED** [review]

File: src/main/java/practice/CandidateValidator.java [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE = 35;
+    private static final String COUNTRY = "Ukrainian";
+    private static final int MIN_LIVE_IN_COUNTRY = 10;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        String[] yearLiveInUkraine = candidate.getPeriodsInUkr().split("-");
+        int periodsInUkraine = Integer.parseInt(yearLiveInUkraine[1])
+                - Integer.parseInt(yearLiveInUkraine[0]);
```
Comment (Sofasmile) [inline src/main/java/practice/CandidateValidator.java:15]: let's create constants for indexes

File: src/main/java/practice/StreamPractice.java [hunk 4]
```diff
@@ -14,7 +19,15 @@ public class StreamPractice {
...
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(n -> Arrays.stream(n.split(","))
+                        .map(Integer::parseInt))
+                .flatMap(Stream::sorted)
+                .filter(i -> i % 2 == 0)
+                .mapToInt(i -> i)
+                .min().orElseThrow(() ->
```
Comment (Sofasmile) [inline src/main/java/practice/StreamPractice.java:28]:
```suggestion
                .min()
                .orElseThrow(() ->
```

File: src/main/java/practice/StreamPractice.java [hunk 5]
```diff
@@ -35,7 +53,9 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
      */
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream().filter(p -> p.getSex() == Person.Sex.MAN
```
Comment (Sofasmile) [inline src/main/java/practice/StreamPractice.java:56]:
```suggestion
        return peopleList.stream()
                    .filter(p -> p.getSex() == Person.Sex.MAN
```

File: src/main/java/practice/StreamPractice.java [hunk 6]
```diff
@@ -50,7 +70,11 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getAge() >= fromAge && ((p.getAge() <= femaleToAge
+                        && p.getSex() == Person.Sex.WOMAN)
```
Comment (Sofasmile) [inline src/main/java/practice/StreamPractice.java:75]: better firstly check sex and then age:
```suggestion
                .filter(p -> p.getAge() >= fromAge && ((p.getSex() == Person.Sex.WOMAN
                        && p.getAge() <= femaleToAge)
```

File: src/main/java/practice/StreamPractice.java [hunk 7]
```diff
@@ -50,7 +70,11 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
...
+                        || (p.getAge() <= maleToAge && p.getSex() == Person.Sex.MAN)))
+                .collect(Collectors.toList());
```
Comment (Sofasmile) [inline src/main/java/practice/StreamPractice.java:77]: same here

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**